lib/WTSI/NPG/Data/ConsentWithdrawn.pm

```diff
@@ -122,63 +123,59 @@ sub process {
 }
 
 # List of files with sample_consent_withdrawn flag set
-has '_files' => (
+has 'files_withdrawn' => (
   isa        => 'ArrayRef',
   is         => 'ro',
   required   => 0,
   lazy_build => 1,
 );
-sub _build__files {
+
+sub _build_files_withdrawn {
   my $self = shift;
 
-  my $iquest_cmd = $self->_iquery;
-  $self->info("Will run: $iquest_cmd");
-  my @files;
-  my $no_rows_error = 0;
-
-  my $pid = open3( undef, my $iquest_out_fh, undef, $iquest_cmd);
-  while (my $line = <$iquest_out_fh> ) {
-    chomp $line;
-    if($line =~ /^ERROR/mxs) {
-      if ($line !~ /CAT_NO_ROWS_FOUND/mxs) {
-        $self->debug($line);
-      } else {
-        $no_rows_error = 1;
-      }
-      last;
-    }
-    if ($line =~/[.]bam$|[.]cram$/mxs) {
-      $self->debug('Found ' . $line);
-      push @files, $line;
-    }
+  my @query;
+  if ($self->zone) {
+    push @query, q[-z], $self->zone;
+  }
+
+  my $select = q[select COLL_NAME, DATA_NAME ] .
+      q[where META_DATA_ATTR_NAME = 'sample_consent_withdrawn' ] .
+      q[and META_DATA_ATTR_VALUE = '1' ] .
+      q[and DATA_NAME not like '%header.bam%'];
+
+  if ($self->collection) {
+      $select .= sprintf q[ and COLL_NAME like '%s%%'], $self->collection();
   }
+  push @query, q[%s/%s], qq("$select");
 
-  waitpid $pid, 0;
-  if( $CHILD_ERROR >> $EXIT_CODE_SHIFT ) {
-    if (!$no_rows_error) {
-      $self->logcroak("Failed: $iquest_cmd");
+  $self->info('Will run: ', join q[ ], @query);
```

Comment:
```suggestion
  $self->info('Will run: iquest', join q[ ], @query);
```
For clarity?

Reply:
Agreed (needs a trailing space after `iquest`)